cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java

```diff
@@ -178,101 +143,30 @@ private void registerGaugeSensors(MetricRegistry dropwizardMetricRegistry) {
         readyBrokers.put(entry.getKey(), Math.max(0, partitionMovementConcurrency - entry.getValue()));
       }
     }
-    return _executionTaskPlanner.getReplicaMovementTasks(readyBrokers, _inProgressPartitions);
-  }
-
-  /**
-   * Returns a list of proposals that move the leadership.
-   */
-  public synchronized List<ExecutionTask> getLeadershipMovementTasks() {
-    return _executionTaskPlanner.getLeadershipMovementTasks(leadershipMovementConcurrency());
-  }
-
-  /**
-   * Returns the remaining partition movement tasks.
-   */
-  public synchronized Set<ExecutionTask> remainingPartitionMovements() {
-    return _executionTaskPlanner.remainingReplicaMovements();
+    return _executionTaskPlanner.getInterBrokerReplicaMovementTasks(readyBrokers, _inProgressPartitionsForInterBrokerMovement);
   }
 
   /**
-   * Returns the remaining leadership movement tasks;
+   * Returns a list of execution tasks that move the replicas cross disks of the same broker.
    */
-  public synchronized Collection<ExecutionTask> remainingLeadershipMovements() {
-    return _executionTaskPlanner.remainingLeadershipMovements();
-  }
-
-  /**
-   * Returns the remaining data to move in MB.
-   */
-  public synchronized long remainingDataToMoveInMB() {
-    return _executionTaskPlanner.remainingDataToMoveInMB();
-  }
-
-  /**
-   * Returns the in execution data to move in MB -- i.e. data to move for in progress or aborting tasks.
-   */
-  public synchronized long inExecutionDataToMoveInMB() {
-    return _inExecutionDataToMove;
-  }
-
-  /**
-   * Get all the tasks that are not completed yet.
-   * The uncompleted tasks include tasks in IN_PROGRESS and ABORTING state.
-   */
-  public synchronized Set<ExecutionTask> inExecutionTasks() {
-    Set<ExecutionTask> inExecution = new HashSet<>();
-    inExecution.addAll(_executionTaskTracker.tasksInState(ExecutionTask.State.IN_PROGRESS));
-    inExecution.addAll(_executionTaskTracker.tasksInState(ExecutionTask.State.ABORTING));
-    return inExecution;
-  }
-
-  /**
-   * Get all the tasks that are not completed yet.
-   * The uncompleted tasks include tasks in IN_PROGRESS and ABORTING state.
-   * @param type the task type to get.
-   */
-  public synchronized Set<ExecutionTask> inExecutionTasks(ExecutionTask.TaskType type) {
-    Set<ExecutionTask> inExecution = new HashSet<>();
-    _executionTaskTracker.tasksInState(ExecutionTask.State.IN_PROGRESS).forEach(t -> {
-      if (t.type() == type) {
-        inExecution.add(t);
-      }
-    });
-    _executionTaskTracker.tasksInState(ExecutionTask.State.ABORTING).forEach(t -> {
-      if (t.type() == type) {
-        inExecution.add(t);
+  public synchronized List<ExecutionTask> getIntraBrokerReplicaMovementTasks() {
```

Comment:
Code duplication: Can we extract the common logic with `getInterBrokerReplicaMovementTasks`?